libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -0,0 +1,614 @@
+        //const float pitch_rate_deg = (int16_t)UINT16_VALUE(_msg_buff[_msg_buff_data_start+9], _msg_buff[_msg_buff_data_start+8]) * 0.1;   // pitch rate
+        //const float roll_rate_deg = (int16_t)UINT16_VALUE(_msg_buff[_msg_buff_data_start+11], _msg_buff[_msg_buff_data_start+10]) * 0.1;  // roll rate
+        break;
+    }
+
+    default:
+        debug("Unhandled CmdId:%u", (unsigned)_parsed_msg.command_id);
+        break;
+    }
+
+    // handle unexpected data buffer lenth
+    if (unexpected_len) {
+        debug("CmdId:%u unexpected len:%u", (unsigned)_parsed_msg.command_id, (unsigned)_parsed_msg.data_bytes_received);
+    }
+}
+
+// methods to send commands to gimbal
+void AP_Mount_Siyi::send_packet(SiyiCommandId cmd_id, const uint8_t* databuff, uint8_t databuff_len)
+{
+    // calculate packet size
+    const uint8_t packet_size = AP_MOUNT_SIYI_PACKETLEN_MIN + databuff_len;
+
+    // check for sufficient space in outgoing buffer
+    if (_uart->txspace() < packet_size) {
+        return;
+    }
+
+    // buffer for holding outgoing packet
+    uint8_t send_buff[packet_size];
+    uint8_t send_buff_len = 0;
+
+    // packet header
+    send_buff[send_buff_len++] = AP_MOUNT_SIYI_HEADER1;
+    send_buff[send_buff_len++] = AP_MOUNT_SIYI_HEADER2;
+
+    // CTRL.  Always request ACK
+    send_buff[send_buff_len++] = 1;
+
+    // Data_len.  protocol supports uint16_t but messages are never longer than 22 bytes
+    send_buff[send_buff_len++] = databuff_len;
+    send_buff[send_buff_len++] = 0;
+
+    // SEQ (sequence)
+    send_buff[send_buff_len++] = LOWBYTE(_last_seq);
+    send_buff[send_buff_len++] = HIGHBYTE(_last_seq++);
+
+    // CMD_ID
+    send_buff[send_buff_len++] = (uint8_t)cmd_id;
+
+    // DATA
+    if (databuff_len != 0) {
+        memcpy(&send_buff[send_buff_len], databuff, databuff_len);
+        send_buff_len += databuff_len;
+    }
+
+    // CRC16
+    const uint16_t crc = crc16_ccitt(send_buff, send_buff_len, 0);
+    send_buff[send_buff_len++] = LOWBYTE(crc);
+    send_buff[send_buff_len++] = HIGHBYTE(crc);
+
+    // send packet
+    _uart->write(send_buff, send_buff_len);
+}
+
+// send a packet with a single data byte to gimbal
+void AP_Mount_Siyi::send_1byte_packet(SiyiCommandId cmd_id, uint8_t data_byte)
+{
+    send_packet(cmd_id, &data_byte, 1);
+}
+
+// rotate gimbal.  pitch_rate and yaw_rate are scalars in the range -100 ~ +100
+// yaw_is_ef should be true if gimbal should maintain an earth-frame target (aka lock)
+void AP_Mount_Siyi::rotate_gimbal(int8_t pitch_scalar, int8_t yaw_scalar, bool yaw_is_ef)
+{
+    // send lock/follow value if it has changed
+    if ((yaw_is_ef != _last_lock) || (_lock_send_counter >= AP_MOUNT_SIYI_LOCK_RESEND_COUNT)) {
+        set_lock(yaw_is_ef);
+        _lock_send_counter = 0;
+        _last_lock = yaw_is_ef;
+    } else {
+        _lock_send_counter++;
+    }
+
+    const uint8_t yaw_and_pitch_rates[] = {(uint8_t)yaw_scalar, (uint8_t)pitch_scalar};
```

Comment:
```suggestion
    const uint8_t yaw_and_pitch_rates[] {(uint8_t)yaw_scalar, (uint8_t)pitch_scalar};
```

Generally try to avoid `=` when you can do brace-initialisation (unless it makes things look *really* ugly, which it can...)

Reply:
ok, removed =